### PR TITLE
[SeaTunnel #836] [sql] Support set operation statements in sql configuration

### LINKED ISSUE #836: [Feature][sql] Support set operation statements in sql configuration
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support set operation statements in sql configuration, like `SET table.dml-sync = true;`

### Usage Scenario

Operation to represent SET command.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leo65535**
## Purpose of this pull request

Fixed #836 

We need to start a standalone flink cluster first, then we can submit sql job in the past, beacuse the sql job is always run asynchronously by default. We can support set operation statements in sql configuration, like `SET table.dml-sync = true;` statement. After that we can run sql job in foreground.

Yeah, support run job in the foreground is just only one of the reasons for introducing this feature.

The keypoint of this feature is to support set operation.

## Run job local without startting cluster.
1. flink home
```
export FLINK_HOME=/work/BigData/install/flink/flink-1.13.3
```

2. application.conf
```
SET table.dml-sync = true;

CREATE TABLE events (
  f_type INT,
  f_uid INT,
  ts AS localtimestamp,
  WATERMARK FOR ts AS ts
) WITH (
  'connector' = 'datagen',
  'rows-per-second'='5',
  'fields.f_type.min'='1',
  'fields.f_type.max'='5',
  'fields.f_uid.min'='1',
  'fields.f_uid.max'='1000'
);

CREATE TABLE print_table (
  type INT,
  uid INT,
  lstmt TIMESTAMP
) WITH (
  'connector' = 'print',
  'sink.parallelism' = '1'
);

INSERT INTO print_table SELECT * FROM events where f_type = 1;
```

3. run local
```
bin/start-seatunnel-sql.sh --target local
```

4. result
```
[dcadmin@dcadmin seatunnel-dist-2.0.4-2.11.8]$ bin/start-seatunnel-sql.sh --target local
Job has been submitted with JobID 6d89f68da4c8f5f3e5c746fc595d44e7
+I[1, 457, 2021-12-20T16:09:06.476]
+I[1, 73, 2021-12-20T16:09:08.470]
+I[1, 736, 2021-12-20T16:09:08.470]
+I[1, 862, 2021-12-20T16:09:11.472]
+I[1, 953, 2021-12-20T16:09:11.472]
+I[1, 278, 2021-12-20T16:09:11.472]
+I[1, 63, 2021-12-20T16:09:13.472]
+I[1, 693, 2021-12-20T16:09:13.473]
```

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by leo65535** [comment]
hi @garyelephant @CalvinKirs, help to review, thanks

File: seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/job/Executor.java [hunk 1]
```diff
@@ -46,17 +48,30 @@ private static StatementSet handleStatements(String workFlowContent, StreamTable
...
                 tEnv.executeSql(stmt);
             }
         }
         return statementSet;
     }
 
+    private static void callSetOperation(Configuration configuration, SetOperation setOperation) {
+
+        // set property
+        if (setOperation.getKey().isPresent() && setOperation.getValue().isPresent()) {
```
Comment (cody1993) [inline seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/job/Executor.java:70]: we can improve here by using `orElseThrow`.

Comment (leo65535) [reply seatunnel-core/seatunnel-core-sql/src/main/java/io/github/interestinglab/waterdrop/core/sql/job/Executor.java:70]: Make sense, thanks.

**Comment by cody1993** [comment]
LGTM, just only one placke need to be modified.

**Review by kezhenxu94 — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**